docs/contributing.rst

```diff
@@ -16,6 +21,92 @@ which tracks the latest release
 (see :doc:`installation` for the basic procedure for running).
 
 
+***********
+Style guide
+***********
+
+*XCP-D* is a Python library, primarily written as Nipype interfaces and workflows.
+As such, we follow a specific coding style adapted to both Python generally and Nipype
+specifically.
+
+*XCP-D* uses a number of ``flake8`` extensions to evaluate coding style,
+but the main ones are ``black`` and ``isort``.
+
+.. tip::
+
+   Please run ``isort`` and ``black`` on the ``XCP-D`` codebase before opening a pull request.
+
+   Also run ``flake8`` to check for any additional style issues.
+
+
+Workflow coding style
+=====================
+
+Workflows should largely be written just like other functions (which should go in ``xcp_d.utils``),
+except for a couple of elements.
+
+First, ``black`` tends to format workflow connections awkwardly, so contributors should add
+``# fmt:off`` before, and ``# fmt:on`` after, each workflow connection statement.
+
+Second, workflow connections should occur shortly after the associated nodes are defined,
```

Comment:
Is this the official niprep way? I tend to do it the opposite in qsiprep

Reply:
I think all of the nipreps repos bundle all of the connections at the end. I just find it harder to read that way, so I figured I'd put my preferred style in this doc.